### PR TITLE
Add run stage option

### LINKED ISSUE #10: Add oci run sub command
Use case:
- Run custom build stages like new network integration tests:
  https://gerrit.ovirt.org/c/98187/

Example usage:

    oci run --stage network-integ CHANGE_NUMBER

The command should build one or more stages using jenkins.run(), and watch the build until it completes.

Complications:
- jenkins.run() accept only single stage now, need to change ti to accept multiple stages.
- Need to check how to pass multiple stages to jenkins. If there is no way to do it, we can support only one stage for now.

### PR DESCRIPTION
**Pull request by bell-lev**
resolves #10

### CODE REVIEW

**Comment by bell-lev** [comment]
network-check run with this patch:
https://jenkins.ovirt.org/job/standard-manual-runner/261/

File: oci/main.py [hunk 1]
```diff
@@ -43,15 +43,66 @@ def run():
...
+    stage_parser.add_argument(
+        'change',
+        help='Gerrit change number')
+
     args = parser.parse_args()
 
     logging.basicConfig(
         level=logging.DEBUG if args.debug else logging.WARNING,
         format="%(asctime)s %(levelname)-7s [%(name)s] %(message)s")
-
```
Comment (nirs) [inline oci/main.py]: Unrelated change

**Review by nirs — CHANGES_REQUESTED** [review]
Nice!

Can you remove the unrelated change?

File: oci/main.py [hunk 2]
```diff
@@ -43,15 +43,66 @@ def run():
              'under "ENGINE_VERSION"',
         default='master')
 
+    stage_parser = subparsers.add_parser(
```
Comment (nirs) [inline oci/main.py]: For consistency with other code, lets call this run_parser.

File: oci/main.py [hunk 3]
```diff
@@ -43,15 +43,66 @@ def run():
...
     args = parser.parse_args()
 
     logging.basicConfig(
         level=logging.DEBUG if args.debug else logging.WARNING,
         format="%(asctime)s %(levelname)-7s [%(name)s] %(message)s")
-
     args.command(args)
 
 
+def run_stage(args):
```
Comment (nirs) [inline oci/main.py]: For consistency with other code, lets call this run()

File: oci/main.py [hunk 4]
```diff
@@ -43,15 +43,66 @@ def run():
...
 
+    stage_parser = subparsers.add_parser(
+        'run',
+        help='run stage for a change')
+    stage_parser.set_defaults(command=run_stage)
+
+    stage_parser.add_argument(
+        '--stage',
+        '-s',
+        help='stage name')
```
Comment (nirs) [inline oci/main.py:53]: This is optional argument without a default. If we run this:

    ./ovirt-ci run 12345

We will fail trying to build a stage None.

I think this should be a positional argument like this:

    ./ovirt-ci run check-network 12345

**Review by nirs — CHANGES_REQUESTED** [review]

File: oci/main.py (hunk 3 above)
Comment (bell-lev) [reply oci/main.py]: Well the thing is the name of the main function is run().

https://github.com/nirs/oci/blob/23a56466ee97c0e2e6a1da4a5ab5f42695f76f04/oci/main.py#L11-L13

Comment (nirs) [reply oci/main.py]: So we can use run_stage_parser.

File: oci/main.py (hunk 4 above)
Comment (bell-lev) [reply oci/main.py:53]: I hope i understood correctly.

File: oci/main.py (hunk 3 above)
Comment (bell-lev) [reply oci/main.py]: Done.

File: oci/main.py (hunk 2 above)
Comment (bell-lev) [reply oci/main.py]: Makes sense. Done.

File: oci/main.py (hunk 1 above)
Comment (bell-lev) [reply oci/main.py]: Done.

**Comment by bell-lev** [comment]
https://jenkins.ovirt.org/job/standard-manual-runner/271/
Ran with the lastest pr

**Comment by nirs** [comment]
Thanks Bell, I push this with minor changes as 296e74404c1f1f8aa3da885d0319d588791c4b5a
